daemon/fqdn.go

```diff
@@ -246,8 +276,10 @@ func (d *Daemon) bootstrapFQDN(restoredEndpoints *endpointRestoreState) (err err
 				if err != nil {
 					log.WithError(err).Error("error updating internal DNS cache for rule generation")
 				}
+				endMetric()
 			}
 
+			stat.ProcessingTime.End(true)
```

Comment:
Why doesn't this need to do `endMetric()`?

Reply:
For that matter, why not just `defer endMetric()` at the start so we don't have to worry about every return condition?